packages/cli/src/apps/wallet/configurator/configurator.ts

```diff
@@ -21,6 +21,7 @@ export interface Config {
   seedKeystore?: EncryptedWallet
   password?: string
   numberOfAccounts?: number
+  snarkKeyCid?: string
```

Comment:
Ohhhh, I like this typing!

Reply:
BTW love to see the idea to use IPFS for snark key download

Reply:
I tried using this type but it doesn't work as expected. TS evaluates it strangely when assigning other types with similar constraints into it. For instance, if I destructure both `snarkKeyCid` and `snarkKeyPath` and assign it onto a `Config` param it will fail because it expects either `snarkKeyPath: string` or `snarkKeyPath: undefined`, but never `snarkKeyPath: string | undefined`.

There's probably a way around it but I don't know if it's worth the headache. We check for either one at runtime and supply a default cid now so this case should never happen.

Reply:
> There's probably a way around it but I don't know if it's worth the headache. We check for either one at runtime and supply a default cid now so this case should never happen.

I totally agree with that. Hope you did not much time to spend on this.